content/recipes/authoring/kata-snippets/description.md

```diff
@@ -5,6 +5,54 @@ prev: /recipes/authoring/kata-snippets/
+
+
+## General
+
+- **Description should hold to general [Content Quality Guidelines](/recipes/authoring/)** where applicable.
+- **Descriptions should be written in English.** Codewars community is international, and not every user, or every author, can use English perfectly. It's fully understandable, and to keep descriptions clear for everyone, authors should ask other users for proofreading, spellchecking, and ask for help if necessary.
+- **All remarks and complaints about instructions not being clear should be considered with proper attention.** Authors should remember that after working on a kata for a long time, they understand it much better than others, and it's easy for them to miss some issues. Users may have completely different point of view, or simply understand things differently. Ask others if description is clear enough, consider their suggestions, and clarify things better where necessary.  
+When questions or requests for improvements, clarifications, or examples pop up particularly often, description should be redacted to address them.
+- **Descriptions should not make translating a kata into another programming languages difficult.** Description is shared between all translations of the kata, and sometimes it has to be modified by translators. If there are many translations, with many such modifications, description merge conflicts can occur, making maintenance of a kata difficult. That's why descriptions should be ideally as language-neutral as possible, and use only abstract terms not specific for any programming language. If a description cannot be kept completely language-neutral, language-specific elements should be structured in a way which helps with resolution of merge conflicts: preferrably in paragraphs or larger blocks, and not many languages mentioned together in one sentence.
+
+
+## Structure
+
+- **Description should be structured into meaningfull parts**, for example:
+  - overwiew, general context, or introduction,
+  - general statement of a task user has to solve,
+  - description of input and output, potentially with examples,
+  - detailed rules and requirements,
+  - additional notes, explanations, technical details, follow-ups.  
+Above is just an example, not all sections are necessary if not applicable, and authors can use another structuring if they like. However, related pieces of information should be held together: explaining inputs at 3-4 different places isn't a good idea.
+- **Descriptions should be consise, but precise.** Description of a task should be possibly short, because long descriptions suggest that task is either not very well defined, or is very complex. Fabularized or themed descriptions are OK as long as they don't get in a way of understanding the problem.
+- It's usually recommended to describe general aspects first, and then go further into details, edge cases, and finally, exceptions. Descriptions structured this way are usually easier to follow.
```

Comment:
No, I meant "inputs which should be handled in some exceptional way", for example,  _"Return index of matching element **or -1 if it could not be found**"_ .

Reply:
Gotcha, that's what I thought but then "edge cases" maybe covered that--I guess not since edge cases are well-formed but out of the ordinary whereas exceptions would be malformed input of some sort. Maybe "malformed input handling" or something to that effect is a bit clearer?

Reply:
Exactly, I avoided "edge cases" because I think they are considered "normal", but on the edge. While malformed inputs could be one case of "exceptional cases" I meant.

Could you please suggest some edit here for this point?